diff --git a/format/BUILD.bazel b/format/BUILD.bazel
index 28ceeb0..3c115a0 100644
--- a/format/BUILD.bazel
+++ b/format/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@aspect_bazel_lib//lib:utils.bzl", bazel_lib_utils = "utils")
 load("@bazel_skylib//:bzl_library.bzl", "bzl_library")
 load("//format/private:formatter_binary.bzl", "BUILTIN_TOOL_LABELS")
 
@@ -31,5 +32,5 @@ bzl_library(
     deps = [
         "@bazel_tools//tools/build_defs/repo:http.bzl",
         "@bazel_tools//tools/build_defs/repo:utils.bzl",
-    ],
+    ] + ["@bazel_tools//tools/build_defs/repo:cache.bzl"] if bazel_lib_utils.is_bazel_7_or_greater() else [],
 )
diff --git a/lint/BUILD.bazel b/lint/BUILD.bazel
index ec76472..eb9933c 100644
--- a/lint/BUILD.bazel
+++ b/lint/BUILD.bazel
@@ -230,6 +230,7 @@ bzl_library(
 bzl_library(
     name = "vale_library",
     srcs = ["vale_library.bzl"],
+    deps = _BAZEL_TOOLS,
 )
 
 bzl_library(
