diff --git a/bzl_library.bzl b/bzl_library.bzl
index a139386..ea5c62b 100644
--- a/bzl_library.bzl
+++ b/bzl_library.bzl
@@ -45,7 +45,7 @@ def _bzl_library_impl(ctx):
         ),
     ]
 
-bzl_library = rule(
+_bzl_library = rule(
     implementation = _bzl_library_impl,
     attrs = {
         "srcs": attr.label_list(
@@ -105,3 +105,20 @@ Example:
   ```
 """,
 )
+
+def bzl_library(name, srcs = [], deps = [], **kwargs):    
+    _bzl_library(
+        name = name,
+        srcs = srcs,
+        deps = deps,
+        **kwargs
+    )
+    
+    if hasattr(native, "starlark_doc_extract"):
+        for i, src in enumerate(srcs):
+            extract_target = "{}.doc_extract{}".format(name, i if i > 0 else "")
+            native.starlark_doc_extract(
+                name = extract_target,
+                src = src,
+                deps = [name],
+            )
